### ISSUE TITLE
Rendering failed due to missing libX11.so.6

### ISSUE THREAD

**Issue by alex1989hu** [bitnami/bitnami-docker-grafana#7]
**Description**

Missing shared (libX11.so.6) object causes image rendering failure.

Snippet of the logfile:
```
grafana t=2020-04-26T08:25:27+0000 lvl=eror msg="Rendering failed." logger=context userId=1 orgId=1 uname=admin error="Rendering failed: Error: Failed to launch chrome!\n/opt/bitnami/grafana/data/plugins/grafana-image-renderer/chrome-linux/chrome: error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n"
```

**Steps to reproduce the issue:**

1. Configure valid data source
2. Install valid dashboard
3. Select a chart from the previously configured dashboard
4. Click on Share
5. Click on Direct link rendered image
7. New page will appear with message *Rendering failed. Check the Grafana server logs for the detailed error message.*

**Describe the results you received:**

Rendering failed.

**Describe the results you expected:**

successful rendering.

**Additional information you deem important (e.g. issue happens only occasionally):**

N/A

**Version**

- Output of `docker version`:

```
N/A
```

- Output of `docker info`:

```
N/A
```

- Output of `docker-compose version` (if applicable):

```
N/A
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
```
docker.io/bitnami/grafana:6.7.3-debian-10-r0
```

**Comment by andresbono** [comment]
Hi @alex1989hu, thanks for opening this issue.

Can you try to install the package manually and try again? To do you you can do the following:

- Exec into the container as root, i.e:

```
docker exec -it --user root grafana bash
```

- Install the package and verify that the `libX11.so.6` library is there:

```
install_packages libx11-6
dpkg-query -L libx11-6
ls -la /usr/lib/x86_64-linux-gnu/libX11.so.6
exit
```

- Restart the container

```
docker restart grafana
```

Please, let us know what are the result of these steps. If they are satisfactory, you may want to create a custom container from the Bitnami one.

**Comment by alex1989hu** [comment]
Hi @andresbono, I did not mention that `grafana` runs in Kubernetes cluster; deployed via Helm Chart: https://github.com/bitnami/charts/tree/master/bitnami/grafana. Meaning that there is no similar command line option available like `docker exec -it --user root`, thus I can not download and install the library.

From https://grafana.com/grafana/plugins/grafana-image-renderer:
> However, Chromium browser depends on certain libraries and if you don't have all of those libraries installed in your system you may encounter errors when trying to render an image. For further details and troubleshooting help, please refer to [Grafana Image Rendering documentation](https://grafana.com/docs/grafana/latest/administration/image_rendering/).

Anyway, I started Docker on my machine and executed your command. After that I found the renderer binary relies on several other libraries:
```bash
root@89f526531a1c:/opt/bitnami/grafana# ldd /opt/bitnami/grafana/data/plugins/grafana-image-renderer/chrome-linux/chrome
	linux-vdso.so.1 (0x00007fff87bd3000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f4e1672b000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f4e1670a000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f4e16700000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f4e165bf000)
	libX11-xcb.so.1 => not found
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f4e16595000)
	libXcomposite.so.1 => not found
	libXcursor.so.1 => not found
	libXdamage.so.1 => not found
	libXext.so.6 => not found
	libXfixes.so.3 => not found
	libXi.so.6 => not found
	libXrender.so.1 => not found
	libXtst.so.6 => not found
	libgobject-2.0.so.0 => not found
	libglib-2.0.so.0 => not found
	libnss3.so => not found
	libnssutil3.so => not found
	libsmime3.so => not found
	libnspr4.so => not found
	libcups.so.2 => not found
	libdbus-1.so.3 => not found
	libXss.so.1 => not found
	libgio-2.0.so.0 => not found
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f4e16550000)
	libXrandr.so.2 => not found
	libasound.so.2 => not found
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f4e163cd000)
	libatk-1.0.so.0 => not found
	libatk-bridge-2.0.so.0 => not found
	libpangocairo-1.0.so.0 => not found
	libpango-1.0.so.0 => not found
	libcairo.so.2 => not found
	libatspi.so.0 => not found
	libgtk-3.so.0 => not found
	libgdk-3.so.0 => not found
	libgdk_pixbuf-2.0.so.0 => not found
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f4e163af000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4e161ee000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f4e20778000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f4e15fea000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f4e15de4000)
	libbsd.so.0 => /usr/lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f4e15dc8000)
```

**Comment by andresbono** [comment]
In that case, I would recommend you to create your custom image based on the bitnami one. Something like the following:

```Dockerfile
FROM bitnami/grafana:latest

USER root

# Install extra packages specified in https://grafana.com/docs/grafana/latest/administration/image_rendering/#troubleshoot-image-rendering
RUN install_packages libx11-6 libx11-xcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrender1 libxtst6 libglib2.0-0 libnss3 libcups2  libdbus-1-3 libxss1 libxrandr2 libgtk-3-0 libgtk-3-0 libasound2

USER 1001
```

Let us know if that works for you.

**Comment by andresbono** [comment]
We are finding other issues related to this plugin, see #8. We might change the approach followed to install and run this plugin.

In any case, if would be great if you try the custom image so we know if it is fully functional and covers your needs for now.

**Comment by alex1989hu** [comment]
@andresbono: It worked! 👍

To being specific, I chose `FROM bitnami/grafana:6.7.3-debian-10-r0` but others are the same as you proposed https://github.com/bitnami/bitnami-docker-grafana/issues/7#issuecomment-620442967

Corresponding log entry:
> grafana t=2020-04-30T06:15:54+0000 lvl=info msg=Rendering logger=rendering renderer=plugin path="d-solo/...

**Comment by andresbono** [comment]
Great, thanks for letting us know!

**Comment by juan131** [comment]
Hi everyone,

I just wanted to share with you that we released a new container [Grafana Image Renderer](https://github.com/bitnami/bitnami-docker-grafana-image-renderer) that you can use to run the rendering service on a different container.

Please also note that `grafana-image-renderer` is not installed by default anymore in the Grafana image. Find more information in the links below:

- https://github.com/bitnami/bitnami-docker-grafana#grafana-image-renderer-plugin
- https://github.com/bitnami/bitnami-docker-grafana#673-debian-10-r28

**Comment by alex1989hu** [comment]
> * https://github.com/bitnami/bitnami-docker-grafana#grafana-image-renderer-plugin

Well done! Will it work with Grafana `v7.0.0`?

**Comment by andresbono** [comment]
Hi @alex1989hu, we haven't tested it with Grafana `v7.0.0` yet.

**Comment by andresbono** [comment]
Hi @alex1989hu, we released Grafana 7.0.0 some days ago, in case you want to check: https://github.com/bitnami/bitnami-docker-grafana/tree/master/7/debian-10
